cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerCapacityConfigFileResolver.java

```diff
@@ -140,8 +183,12 @@ private static boolean isJBOD(Map<Resource, Object> brokerCapacity) {
     return brokerCapacity.get(Resource.DISK) instanceof Map;
   }
 
+  private static boolean hasNumCores(Map<Resource, Object> brokerCapacity) {
```

Comment:
This method does not check what the name suggests, it is based on an assumption that if it is a map, the map contains only one key and the key is `num.cores`

Reply:
Added `getUserSpecifiedNumCores(Map<Resource, Object>)` to make the logic more clear.

`... the map contains only one key ...`: This, I suspect, would make the sanity check stricter than intended -- i.e. the map can have other entries without violating our requirements.